### DISCUSSION TITLE
[Proposal]TLS Acceleration with Multi-buffer

### DISCUSSION THREAD

**Discussion by zhoudan-intel** [oceanbase/oceanbase#955] [Ideas]
The OB Cluster can be configured to use secured communication with SSL/TLS. Because SSL/TLS is computation-intensive work, it will be helpful to accelerate TLS.

We'd suggest accelerating TLS with Multi-buffer. It is simple to use Intel QAT OpenSSL Engine to enable Multi-buffer features.
- Intel QAT OpenSSL Engine supports acceleration for both hardware as well as optimized software based on vectorized instructions. 
- Intel QAT OpenSSL Engine supports Multi-buffer based software acceleration for asymmetric PKE algorithms using the Intel Crypto Multi-buffer library.
- Intel QAT OpenSSL Engine supports software-based acceleration for AES-GCM via the Intel Multi-buffer Crypto for IPsec Library.    

Please see more details about the performance improvement of multi-buffer here.
https://www.intel.com/content/www/us/en/architecture-and-technology/crypto-acceleration-in-xeon-scalable-processors-wp.html

**Comment by hnwyllmm** [comment]


**Reply by lzha111** [reply]
@hnwyllmm  do you have more detail about the file encryption? thank you!
